src/System.Net.Sockets/tests/FunctionalTests/SocketOptionNameTest.cs

```diff
@@ -284,6 +284,52 @@ public void ReuseAddress_Windows(bool? exclusiveAddressUse, bool? firstSocketReu
             ReuseAddress(exclusiveAddressUse, firstSocketReuseAddress, secondSocketReuseAddress, expectFailure);
         }
 
+        [Fact]
+        public void BindDuringTcpWait()
+        {
+            int port = 0;
+            using (Socket a = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            {
+                a.Bind(new IPEndPoint(IPAddress.Loopback, 0));
+                port = (a.LocalEndPoint as IPEndPoint).Port;
+                a.Listen(10);
+
+                // Connect a client
+                using (Socket client = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+                {
+                    client.Connect(new IPEndPoint(IPAddress.Loopback, port));
+                    using (Socket acceptedClient = a.Accept())
+                    { }
```

Comment:
a.Accept().Dispose();

Reply:
Should we be explicitly setting a linger time rather than relying on whatever the OS' default is?

Reply:
👍 Yes, some OSes will block close during the linger time.